src/main/java/com/exasol/adapter/dialects/sqlserver/SqlServerIdentifier.java

```diff
@@ -0,0 +1,62 @@
+package com.exasol.adapter.dialects.sqlserver;
+
+import java.util.Objects;
+
+import com.exasol.db.Identifier;
+
+/**
+ * Represents an identifier in the Sql Server database.
+ */
+public class SqlServerIdentifier implements Identifier {
+    private final String id;
+
+    private SqlServerIdentifier(final String id) {
+        this.id = id;
+    }
+
+    /**
+     * Get the quoted identifier as a {@link String}.
+     *
+     * @return quoted identifier
+     */
+    @Override
+    public String quote() {
+        return "[" + this.id + "]";
+    }
+
+    /**
+     * Create a new {@link SqlServerIdentifier}.
+     *
+     * @param id the identifier as {@link String}
+     * @return new {@link SqlServerIdentifier} instance
+     */
+    public static SqlServerIdentifier of(final String id) {
+        if (validate(id)) {
+            return new SqlServerIdentifier(id);
+        } else {
+            throw new AssertionError("E-ID-3: Unable to create identifier \"" + id //
+                    + "\" because it contains illegal characters." //
+                    + " For information about valid identifiers, please refer to" //
+                    + " https://docs.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?redirectedfrom=MSDN&view=sql-server-ver15");
```

Comment:
I would not include the link in the error message, because it might become outdated. Instead, we should point out the problematic characters and move the link to `applyQuote`.

Reply:
I will also place a link in the applyQuote, but I would leave the link in the exception message too for now, because it's hard to describe all the rules in the error message.  I think users should read the documentation for it